File: src/main/java/ru/tykvin/hermes/file/model/FileInfo.java
package ru.tykvin.hermes.file.model;

import lombok.Data;
import org.hibernate.validator.internal.util.logging.formatter.DurationFormatter;

import java.time.Duration;
import java.util.UUID;

@Data
public class FileInfo {
    private final UUID bindingId;
    private final UUID fileId;
    private final String fileName;
    private final Duration lifeTime;
    private final Integer downloadsLeft;
    private final long size;
    private final String url;


    public String getLifeTimeAString() {
        return new DurationFormatter(lifeTime).toString();
    }
}

File: src/main/kotlin/ru/tykvin/hermes/file/controller/FileInfo.kt
package ru.tykvin.hermes.file.controller

import java.time.Duration
import java.util.*

data class FileInfo(
        val bindingId: UUID,
        val fileId: UUID,
        val fileName: String,
        val lifeTime: Duration,
        val downloadsLeft: Int,
        val size: Long,
        val url: String
)
